e2e/cypress/integration/channel/message_parse_spec.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message', () => {
+    before(() => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('MM-17444 - correctly parses "://///" as Markdown and does not break the channel', () => {
+        // # Go to Town Square as test channel
+        cy.get('#sidebarItem_town-square').click({force: true});
+
+        // # Validate if the channel has been opened
+        cy.url().should('include', '/channels/town-square');
+
+        // # type in the message "://///"
+        const message = '://///';
+        const testAfterParsed = message.substr(2);
+        cy.postMessage(message);
+
+        // # check if message sent correctly, it should parse it as 😕////"
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).children().find('span').last().should('have.class', 'emoticon').and('be.visible');
+            cy.get(`#postMessageText_${postId}`).should('be.visible').and('have.text', testAfterParsed);
```

Comment:
Please simplify so that ``cy.get(`#postMessageText_${postId}`)`` is called once.  (nit) Also add check to title.

```javascript
cy.get(`#postMessageText_${postId}`).should('be.visible').within((el) => {
    cy.wrap(el).should('have.text', testAfterParsed);
    cy.get('.emoticon').should('be.visible').and('have.attr', 'title', ':confused:');
});
```

This can be wrapped into a function so it can be used after reload.

Reply:
is this what you mean?

Reply:
I meant:
```javascript
cy.getLastPostId().then((postId) => {
    verifyPostedMessage(postId, textAfterParsed);

    // # check if message still correctly parses after reload
    cy.reload();
    verifyPostedMessage(postId, textAfterParsed);
});
```

If you have,
```javascript
function verifyPostedMessage(postId, text) {
    cy.get(`#postMessageText_${postId}`).should('be.visible').within((el) => {
        cy.wrap(el).should('have.text', text);
        cy.get('.emoticon').should('be.visible').and('have.attr', 'title', ':confused:');
    });
}
```

Reply:
ah. yes. this refactoring makes it clean